### ISSUE TITLE
[connector/failoverconnector] Failing unit test: TestTracesWithFailoverRecovery

### ISSUE THREAD

**Issue by crobert-1** [open-telemetry/opentelemetry-collector-contrib#31005]
### Component(s)

connector/failover

### Describe the issue you're reporting

[CI/CD failing on unrelated PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/7745537973/job/21121803104?pr=30927)

Failure output:
```
--- FAIL: TestTracesWithFailoverRecovery (0.10s)
    traces_test.go:190: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/connector/failoverconnector/traces_test.go:190
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestTracesWithFailoverRecovery
FAIL
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/connector/failoverconnector	0.325s
FAIL
```

**Comment by crobert-1** [comment]
[Possible duplicate, but different test failing](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/7745571276/job/21121919305?pr=30887):
```
--- FAIL: TestLogsWithFailoverRecovery (0.10s)
    logs_test.go:186: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/connector/failoverconnector/logs_test.go:186
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestLogsWithFailoverRecovery
FAIL
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/connector/failoverconnector	0.327s
FAIL
```
It might be worth opening a separate issue for this, but since they look so similar we'll keep together for now.

**Comment by akats7** [comment]
Yep thats a duplicate, let me remove those tests temporarily while I mock out the time dependency, looks like its failing on certain runs

**Comment by codeboten** [comment]
This appears fixed by https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31006

**Comment by crobert-1** [comment]
That fixed the traces failure, but the logs failure still needs addressed: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e0a18f0be7e3e22da86af1246fdbec6f5b00c5b3/connector/failoverconnector/logs_test.go#L139

It looks like the same solution can be used there.

**Comment by akats7** [comment]
> That fixed the traces failure, but the logs failure still needs addressed:
> 
> https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e0a18f0be7e3e22da86af1246fdbec6f5b00c5b3/connector/failoverconnector/logs_test.go#L139
> 
> It looks like the same solution can be used there.

Opening a PR

**Comment by codeboten** [comment]
Thanks @akats7

**Comment by songy23** [comment]
Fixed in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31007

**Comment by crobert-1** [comment]
> Fixed in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31007

The referenced PR only skips the tests, we probably want to leave this issue open until there's an opportunity to fully resolve the issue, or fix the test. Let me know if I missed something, or if that's not the right process. 👍
